frontend/src/components/global/AlgoTextarea.vue

```diff
@@ -18,6 +26,41 @@ export default {
             this.$nextTick(() => {
                 event.target.selectionStart = event.target.selectionEnd = selectionStart + 1;
             });
+
+            this.$emit("change-specific", {
+                start: event.target.selectionStart,
+                end: event.target.selectionStart + 1,
+                size: 1,
+                deltaLineCount: 0,
+            });
+        },
+
+        handleBeforeinput(event) {
+            oldText = event.target.value;
+        },
+
+        handleInput(event) {
+            let newText = event.target.value;
+
+            let oldLen = oldText.length;
+            let newLen = newText.length;
+
+            let lengthDifference = newLen - oldLen;
+            let differenceStart = 0;
+
```

Comment:
Jak można wydzielić jakiś mały spójny kawałek kodu, który realizuje konkretne zadanie to warto to wydzielać do oddzielnej funkcji (nawet jeżeli główna funkcja nie jest zbyt duża tak jak w tym przypadku). Widząc linijkę let differenceStart = findFirstDifferentChar(oldText, newText); co ten fragment robi (i w razie potrzeby wyliczenia tego samego w innym miejscu projektu mamy już gotową funkcję).